File: inheritance.cpp
#include<iostream>
using namespace std;
class Vehicles{
	protected:
		int mSpeed = 0;
	private:
		string color = "yellow";
		bool situation = false;
	public:
		void run(int speed){
			situation = true;
			if(situation == true)
				mSpeed = speed;
		}
		void stop(){
			if(situation == true){
				mSpeed = 0;
				situation = false;
			}
		}
		void show(){
			cout << "Running = " << situation << "\nSpeed = " << mSpeed << "\nColor = " << color << endl;
		}
};
class Taxi : public Vehicles{
	public:
		void fastDrive(){
			mSpeed += 50;
			cout << "Instantaneous speed = " << mSpeed << endl;
		}
};
class Truck : public Vehicles{};
int main(){
	Taxi opel;
	Truck bmc;
	cout << "Car\n";
	opel.run(30);
	opel.show();
	opel.fastDrive();
	cout << "Truck\n";
	bmc.run(20);
	bmc.show();
	bmc.stop();
	bmc.show();
}

File: inheritance.java
public class inheritance {

	public static class Vehicles{
		protected int mSpeed = 0;
		private String color = "yellow";
		private boolean situation = false;
	public void run(int speed) {
		situation = true;
		if(situation == true)
			mSpeed = speed;
	}
	public void stop() {
		if(situation == true) {
			mSpeed = 0;
			situation = false;
		}
	}
	public void show() {
		System.out.println("Running = " + situation + "\nSpeed = " + mSpeed
				+ "\nColor = " + color);
	}
	public class Taxi extends Vehicles{
		public void fastRun() {
			mSpeed += 50;
			System.out.println("Instantaneous = " + mSpeed);
		}
	}
	public class Truck extends Vehicles{}	
	
	public static void main(String[] args) {
		Taxi opel = new Vehicles().new Taxi();
		Truck bmc = new Vehicles().new Truck();
		System.out.println("Car");
		opel.run(50); opel.show(); opel.fastRun(); opel.show();
		System.out.println("Truck");
		bmc.run(30); bmc.show(); bmc.stop(); bmc.show();
		
	}
	}
}
